tasks/collect-pyxis-params/collect-pyxis-params.yaml

```diff
@@ -0,0 +1,49 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: collect-pyxis-params
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that collects pyxis configuration options from the data file
+  params:
+    - name: dataPath
+      type: string
+      description: Path to the JSON string of the merged data containing the pyxis configuration options to use
+      default: data.json
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot spec json file resides
+  results:
+    - name: server
+      type: string
+      description: The server type to use. Options are 'production' and 'stage'. Default is 'production'
+    - name: secret
+      type: string
+      description: |
+        The kubernetes secret to use to authenticate to Pyxis. It needs to contain two keys: key and cert
+  steps:
+    - name: collect-pyxis-params
+      image:
+        quay.io/hacbs-release/release-utils:48a1cd6c939eda25382472135be4afe2c9b4b703
+      script: |
+        #!/usr/bin/env sh
+        set -x
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        if [ $(jq '.pyxis | has("secret")' $DATA_FILE) == false ] ; then
+            echo "Pyxis key or secret missing in data JSON file"
+            exit 1
+        fi
+        jq -r '.pyxis.secret' $DATA_FILE > $(results.secret.path)
+        jq -r '.pyxis.server // "production"' $DATA_FILE > $(results.server.path)
```

Comment:
Not sure if this should be here or in the Pyxis task. This could emit an empty result and the Pyxis one would decide the default. The only thing that we really have to pass is the secret, isn't it?

Reply:
It could go there... I was trying to create a task to collect the things we currently pass as parameters and not have to change the tasks. That was my intent. The server could be determined in each of the pyxis tasks (as long as we ensure to keep them in sync)

Reply:
Let's leave it like this and we rethink it in the future.